CMakeLists.txt

```diff
@@ -481,12 +481,8 @@ endif(HELICS_ENABLE_MPI_CORE)
 # -------------------------------------------------------------
 # add threading support
 # -------------------------------------------------------------
-if(NOT WIN32)
-    set(THREADS_PREFER_PTHREAD_FLAG ON)
-elseif(MSYS OR CYGWIN)
+if(MSYS OR CYGWIN OR NOT WIN32)
     set(THREADS_PREFER_PTHREAD_FLAG ON)
-else()
-    set(CMAKE_USE_WIN32_THREADS_INIT ON)
```

Comment:
Do we know why we set this in the first place.  It is possible it was dealing with something in older cmake versions?

Reply:
There's no useful info about why it was added; the first time it appeared was clang-tidy PR https://github.com/GMLC-TDC/HELICS/pull/386 as part of a much larger commit called "some updates to cmake and requested fixes".

Reply:
The closest conditions surrounding when it gets set in https://github.com/Kitware/CMake/blame/master/Modules/FindThreads.cmake haven't changed significantly in about the past decade.

Setting the variable ourselves doesn't have any affect on what FindThreads.cmake does, and we never reference it in our code either.